simulation/agent.py

```diff
@@ -179,11 +187,50 @@ def _update_sprint(self):
         #Check, if the agent has rested for enough -> ensure, that rests when if can't sprint
         if self.is_resting:
             self.move_speed = 0
+            
+    def update_tower(self):
```

Comment:
if the method is not supposed to be called by the actual agent implementations it should be private, i.e. `_update_tower`
(and a nitpick: you're not updating the tower, but the agent entering the tower, so it's a bit confusing, naming is important ;) )